Backup & Restore with BR fail

## Bug Report

**What version of Kubernetes are you using?**
<!-- You can run `kubectl version` -->

**What version of TiDB Operator are you using?**
<!-- You can run `kubectl exec -n tidb-admin {tidb-controller-manager-pod-name} -- tidb-controller-manager -V` -->
v1.1.9
**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**
<!-- You can run `kubectl get sc` and `kubectl get pvc -n {tidb-cluster-namespace}` -->

**What's the status of the TiDB cluster pods?**
<!-- You can run `kubectl get po -n {tidb-cluster-namespace} -o wide` -->

**What did you do?**
<!-- If possible, provide a recipe for reproducing the error. How you installed tidb-operator and tidb-cluster. -->
Backup with BR with TiDB v4.0.9 without configuring `spec.from`, and TLS is enabled and `${cluster_name}-tidb-client-secret"` is not created.
Restore with BR with TiDB v4.0.9 without configuring `spec.to`, and TLS is enabled.
**What did you expect to see?**
Backup & Restore succeeds
**What did you see instead?**
Error with backup:
```
MountVolume.SetUp failed for volume "tidb-client-tls" : secret "${cluster_name}-tidb-client-secret" not found
```

Error with restore：
```
E0105 15:22:29.000946       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 756 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1a9d9c0, 0x2fa4210)
	k8s.io/apimachinery@v0.0.0/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.0.0/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1a9d9c0, 0x2fa4210)
	runtime/panic.go:679 +0x1b2
github.com/pingcap/tidb-operator/pkg/backup/restore.(*restoreManager).makeRestoreJob(0xc000692f00, 0xc0005f4f00, 0x1d29edb, 0x8, 0x20f2ca0, 0xc000123360, 0x1a)
	github.com/pingcap/tidb-operator@/pkg/backup/restore/restore_manager.go:364 +0xc55
github.com/pingcap/tidb-operator/pkg/backup/restore.(*restoreManager).syncRestoreJob(0xc000692f00, 0xc0005f4f00, 0x1d28655, 0x7)
	github.com/pingcap/tidb-operator@/pkg/backup/restore/restore_manager.go:130 +0x7ea
github.com/pingcap/tidb-operator/pkg/backup/restore.(*restoreManager).Sync(0xc000692f00, 0xc0005f4f00, 0x1d2ac3b, 0x8)
	github.com/pingcap/tidb-operator@/pkg/backup/restore/restore_manager.go:48 +0x35
github.com/pingcap/tidb-operator/pkg/controller/restore.(*defaultRestoreControl).UpdateRestore(0xc00096a6b0, 0xc0005f4f00, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/controller/restore/restore_control.go:49 +0xda
github.com/pingcap/tidb-operator/pkg/controller/restore.(*Controller).syncRestore(...)
	github.com/pingcap/tidb-operator@/pkg/controller/restore/restore_controller.go:133
github.com/pingcap/tidb-operator/pkg/controller/restore.(*Controller).sync(0xc000541a70, 0xc000f739c0, 0x16, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/controller/restore/restore_controller.go:129 +0x1f0
github.com/pingcap/tidb-operator/pkg/controller/restore.(*Controller).processNextWorkItem(0xc000541a70, 0x0)
	github.com/pingcap/tidb-operator@/pkg/controller/restore/restore_controller.go:93 +0x106
github.com/pingcap/tidb-operator/pkg/controller/restore.(*Controller).worker(0xc000541a70)
	github.com/pingcap/tidb-operator@/pkg/controller/restore/restore_controller.go:81 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00144c100)
	k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00144c100, 0x3b9aca00, 0x0, 0x1, 0xc0009e0180)
	k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc00144c100, 0x3b9aca00, 0xc0009e0180)
	k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:88 +0x4d
created by github.com/pingcap/tidb-operator/pkg/controller/restore.(*Controller).Run
	github.com/pingcap/tidb-operator@/pkg/controller/restore/restore_controller.go:73 +0x1d9
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x16ebf25]
goroutine 756 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.0.0/pkg/util/runtime/runtime.go:55 +0x105
panic(0x1a9d9c0, 0x2fa4210)
	runtime/panic.go:679 +0x1b2
github.com/pingcap/tidb-operator/pkg/backup/restore.(*restoreManager).makeRestoreJob(0xc000692f00, 0xc0005f4f00, 0x1d29edb, 0x8, 0x20f2ca0, 0xc000123360, 0x1a)
	github.com/pingcap/tidb-operator@/pkg/backup/restore/restore_manager.go:364 +0xc55
github.com/pingcap/tidb-operator/pkg/backup/restore.(*restoreManager).syncRestoreJob(0xc000692f00, 0xc0005f4f00, 0x1d28655, 0x7)
	github.com/pingcap/tidb-operator@/pkg/backup/restore/restore_manager.go:130 +0x7ea
github.com/pingcap/tidb-operator/pkg/backup/restore.(*restoreManager).Sync(0xc000692f00, 0xc0005f4f00, 0x1d2ac3b, 0x8)
	github.com/pingcap/tidb-operator@/pkg/backup/restore/restore_manager.go:48 +0x35
github.com/pingcap/tidb-operator/pkg/controller/restore.(*defaultRestoreControl).UpdateRestore(0xc00096a6b0, 0xc0005f4f00, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/controller/restore/restore_control.go:49 +0xda
github.com/pingcap/tidb-operator/pkg/controller/restore.(*Controller).syncRestore(...)
	github.com/pingcap/tidb-operator@/pkg/controller/restore/restore_controller.go:133
github.com/pingcap/tidb-operator/pkg/controller/restore.(*Controller).sync(0xc000541a70, 0xc000f739c0, 0x16, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/controller/restore/restore_controller.go:129 +0x1f0
github.com/pingcap/tidb-operator/pkg/controller/restore.(*Controller).processNextWorkItem(0xc000541a70, 0x0)
	github.com/pingcap/tidb-operator@/pkg/controller/restore/restore_controller.go:93 +0x106
github.com/pingcap/tidb-operator/pkg/controller/restore.(*Controller).worker(0xc000541a70)
	github.com/pingcap/tidb-operator@/pkg/controller/restore/restore_controller.go:81 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00144c100)
	k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00144c100, 0x3b9aca00, 0x0, 0x1, 0xc0009e0180)
	k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc00144c100, 0x3b9aca00, 0xc0009e0180)
	k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:88 +0x4d
created by github.com/pingcap/tidb-operator/pkg/controller/restore.(*Controller).Run
	github.com/pingcap/tidb-operator@/pkg/controller/restore/restore_controller.go:73 +0x1d9
```

--- a/pkg/backup/backup/backup_manager.go
+++ b/pkg/backup/backup/backup_manager.go
@@ -383,10 +383,10 @@ func (bm *backupManager) makeBackupJob(backup *v1alpha1.Backup) (*batchv1.Job, s
 		})
 	}
 
-	if tc.Spec.TiDB.TLSClient != nil && tc.Spec.TiDB.TLSClient.Enabled && !tc.SkipTLSWhenConnectTiDB() {
+	if backup.Spec.From != nil && tc.Spec.TiDB.TLSClient != nil && tc.Spec.TiDB.TLSClient.Enabled && !tc.SkipTLSWhenConnectTiDB() {
 		args = append(args, "--client-tls=true")
 		clientSecretName := util.TiDBClientTLSSecretName(backup.Spec.BR.Cluster)
-		if backup.Spec.From != nil && backup.Spec.From.TLSClientSecretName != nil {
+		if backup.Spec.From.TLSClientSecretName != nil {
 			clientSecretName = *backup.Spec.From.TLSClientSecretName
 		}
 		volumeMounts = append(volumeMounts, corev1.VolumeMount{

--- a/pkg/backup/restore/restore_manager.go
+++ b/pkg/backup/restore/restore_manager.go
@@ -358,7 +358,7 @@ func (rm *restoreManager) makeRestoreJob(restore *v1alpha1.Restore) (*batchv1.Jo
 		})
 	}
 
-	if tc.Spec.TiDB.TLSClient != nil && tc.Spec.TiDB.TLSClient.Enabled && !tc.SkipTLSWhenConnectTiDB() {
+	if restore.Spec.To != nil && tc.Spec.TiDB.TLSClient != nil && tc.Spec.TiDB.TLSClient.Enabled && !tc.SkipTLSWhenConnectTiDB() {
 		args = append(args, "--client-tls=true")
 		clientSecretName := util.TiDBClientTLSSecretName(restore.Spec.BR.Cluster)
 		if restore.Spec.To.TLSClientSecretName != nil {

